hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -170,10 +182,17 @@ public void scan(Option<List<String>> keys) {
     HoodieTimeline completedInstantsTimeline = commitsTimeline.filterCompletedInstants();
     HoodieTimeline inflightInstantsTimeline = commitsTimeline.filterInflights();
     try {
-      // iterate over the paths
+
+      // If virtual keys are enabled, set the key field accordingly.
+      String keyField = HoodieRecord.RECORD_KEY_METADATA_FIELD;
+      if (virtualKeysEnabled) {
```

Comment:
single line ternary expression? of all the places, this seems the most apt. 

```
final String keyField = virtualKeysEnabled ? this.simpleKeyGenFields.get().getKey() : HoodieRecord.RECORD_KEY_METADATA_FIELD;
```

Reply:
fixed.